### PR TITLE
[client] add support for registering validator

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

Support registering validator script in client. It's gated by mint_key now, but wonder if we should open this script the same as transfer.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

test in dev net, register a validator not in genesis and add it to the network. will follow up with a smoke test.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

**Comment by sblackshear** [comment]
>  It's gated by mint_key now, but wonder if we should open this script the same as transfer.

It looks like the (pre-existing) `add_validator` is gated by the mint key, but this new `register_validator` script is not. Both of these choices make sense to be; just wanted to make sure I understand correctly.

**Comment by zekun000** [comment]
@sblackshear by `gated by mint_key` I mean the command will only appear with it's launched with the mint_key i.e. it's part of DevCommand, the txn is of course authorized by the validator's own private key.

all the smoke tests are using client commands, I'm going to add one for the register_validator too(need some change how the swarm generate the genesis)

File: client/cli/src/client_proxy.rs [hunk 1]
```diff
@@ -343,6 +347,53 @@ impl ClientProxy {
         }
     }
 
+    /// Register an account as validator candidate with ValidatorConfig
+    pub fn register_validator(
```
Comment (davidiw) [inline client/cli/src/client_proxy.rs]: wow, I might suggest at this point we generate new config file that represents the public facing config for a node. The following looks hard to maintain and scary...

Comment (davidiw) [reply client/cli/src/client_proxy.rs]: I could punt on this is if you gave a really good example in the document section above, but this isn't really reusable whereas being able to point to a config file is.

File: client/cli/src/dev_commands.rs [hunk 2]
```diff
@@ -216,3 +217,29 @@ impl Command for DevCommandGenWaypoint {
...
 }
+
+pub struct DevCommandRegisterValidator {}
+
+impl Command for DevCommandRegisterValidator {
+    fn get_aliases(&self) -> Vec<&'static str> {
+        vec!["register_validator"]
+    }
+    fn get_params_help(&self) -> &'static str {
+        "<validator_account_address> <validator_account_private_key> <consensus_public_key> <network_signing_key> <network_identity_key> <network_address> <fullnode_identity_key> <fullnode_network_address>"
```
Comment (davidiw) [inline client/cli/src/dev_commands.rs]: hahaha

**Comment by zekun000** [comment]
I'm not a big fan of this too, I can turn it to accept a `node.config.toml` later

**Pull request closed without merging**